packages/edit-site/src/components/template-switcher/index.js

```diff
@@ -70,52 +69,64 @@ export default function TemplateSwitcher( {
 		setThemePreviewVisible( () => false );
 	};
 
+	const registry = useRegistry();
+	const [ homeId, setHomeId ] = useState();
+
+	useEffect( () => {
+		findTemplate(
+			'/',
+			registry.__experimentalResolveSelect( 'core' ).getEntityRecords
+		).then(
+			( newHomeId ) => setHomeId( newHomeId ),
+			() => setHomeId( null )
+		);
```

Comment:
It's great that we now have the `findTemplate` util, but TBH, its signature (specifically, the need to pass in a selector as an argument) and async nature (requiring local component state) are among the reasons why I was opting for moving this kind of stuff into resolvers.

Not relevant for this PR, but I wanted to point it out in case we'd be willing to reconsider later.

Reply:
We could wrap it in a custom hook too.

Reply:
Resolvers need to be tied to the store state, and it would be awkward to have these temporary values there.

Reply:
I wouldn't introduce a dedicated resolver for `findTemplate`; I was specifically thinking of one for `getTemplateParts( path )` that would call `const templateId = yield findTemplate( path )` internally, and use that to fetch the template parts for that template.

Or we could even modify the entity endpoints for templates and template parts some more to accept a `path` arg, and do the template lookup on the server side.

Reply:
>Or we could even modify the entity endpoints for templates and template parts some more to accept a path arg, and do the template lookup on the server side.

That makes more sense don't you think?

Reply:
Possibly, yeah. I'm not sure if it'd maybe bend the entities abstraction a bit much (if we keep adding rather specific custom args to it that change the query significantly), but maybe that's alright.

Think it still makes sense to merge _this_ PR as a first step towards fetching data from the REST API (rather than from a server-passed JS variable)? :thinking:

Reply:
I'd rather avoid adding `templateId` as a parameter only to remove it later.

Reply:
Makes sense.

Unfortunately, doing the path-to-template resolution on the server side will be non-trivial AFAICS -- the [`get_{$type}_template()` functions](https://core.trac.wordpress.org/browser/tags/5.4/src/wp-includes/template.php) don't accept any args to pass entity specifiers (e.g. post/page IDs, category slugs, etc) but infer those from context (e.g. via [`get_queried_id()`](https://core.trac.wordpress.org/browser/tags/5.4/src/wp-includes/template.php#L426)). _Maybe_ it'll be possible to hook a filter into those, but there's a chance that we'll end up replicating a lot of that logic :slightly_frowning_face:

Reply:
Let's leave it for later then.